src/identity-providers/add/DescriptorSettings.tsx

```diff
@@ -0,0 +1,428 @@
+                  }
+                  isOpen={signatureAlgorithmDropdownOpen}
+                  onSelect={(_, value) => {
+                    onChange(value as string);
+                    setSignatureAlgorithmDropdownOpen(false);
+                  }}
+                  selections={value}
+                  variant={SelectVariant.single}
+                >
+                  <SelectOption key={0} value="RSA_SHA1" />
+                  <SelectOption key={1} value="RSA_SHA256" isPlaceholder />
+                  <SelectOption key={2} value="RSA_SHA256_MGF1" />
+                  <SelectOption key={3} value="RSA_SHA512" />
+                  <SelectOption key={4} value="RSA_SHA512_MGF1" />
+                  <SelectOption key={5} value="DSA_SHA1" />
+                </Select>
+              )}
+            ></Controller>
+          </FormGroup>
+          <FormGroup
+            label={t("samlSignatureKeyName")}
+            labelIcon={
+              <HelpItem
+                helpText={th("samlSignatureKeyName")}
+                forLabel={t("samlSignatureKeyName")}
+                forID="kc-samlSignatureKeyName"
+              />
+            }
+            fieldId="kc-samlSignatureKeyName"
+          >
+            <Controller
+              name="config.xmlSigKeyInfoKeyNameTransformer"
+              defaultValue="keyID-option"
+              control={control}
+              render={({ onChange, value }) => (
+                <Select
+                  toggleId="kc-samlSignatureKeyName"
+                  onToggle={() =>
+                    setSamlSignatureKeyNameDropdownOpen(
+                      !samlSignatureKeyNameDropdownOpen
+                    )
+                  }
+                  isOpen={samlSignatureKeyNameDropdownOpen}
+                  onSelect={(_, value) => {
+                    onChange(value as string);
+                    setSamlSignatureKeyNameDropdownOpen(false);
+                  }}
+                  selections={value}
+                  variant={SelectVariant.single}
+                >
+                  <SelectOption key={0} value="NONE" />
+                  <SelectOption key={1} value={t("keyID")} isPlaceholder />
+                  <SelectOption key={2} value={t("certSubject")} />
+                </Select>
+              )}
+            ></Controller>
+          </FormGroup>
+        </>
+      )}
+
+      <SwitchField
+        field="config.wantAssertionsSigned"
+        label="wantAssertionsSigned"
+        isReadOnly={readOnly}
+      />
+
+      <SwitchField
+        field="config.wantAssertionsEncrypted"
+        label="wantAssertionsEncrypted"
+        isReadOnly={readOnly}
+      />
+      <SwitchField
+        field="config.forceAuthn"
+        label="forceAuthentication"
+        isReadOnly={readOnly}
+      />
+
+      <SwitchField
+        field="config.validateSignature"
+        label="validateSignature"
+        isReadOnly={readOnly}
+      />
+      {validateSignature === "true" && (
```

Comment:
If the aforementioned changes are implemented to make this a boolean there is no need to compare this value to a string anymore.

```suggestion
      {validateSignature && (
```

Reply:
I'm not changing these because they are stored as text in the API and changing them to booleans ruins the show/hide on load.